### PR TITLE
add integration test framework

### PR DESCRIPTION
**Pull request by sonyafenge**

### CODE REVIEW

File: test/integration/service-api/basic_test.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+var singleTestLock = sync.Mutex{}
+
+func setUp() *distributor.ResourceDistributor {
+	singleTestLock.Lock()
+	return distributor.GetResourceDistributor()
+}
+
+func tearDown(resourceDistributor *distributor.ResourceDistributor) {
+	defer singleTestLock.Unlock()
```
Comment (Sindica) [inline test/integration/service-api/basic_test.go:52]: Why defer?

File: test/integration/service-api/basic_test.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	for i := 0; i < eventNum; i++ {
+		rvToGenerate += 1
+		node := createRandomNode(rvToGenerate)
+		nodeEvent := event.NewNodeEvent(node, event.Added)
+		result[i] = nodeEvent
+	}
+	return result
+}
+
+func TestHttpGet(t *testing.T) {
```
Comment (Sindica) [inline test/integration/service-api/basic_test.go:79]: I am confused. What is the difference between this test and TestHttpGet in installer_test.go?

**Comment by yb01** [comment]
should be an e2e test with verification of pipe line results. should be OK to main directly.

**Comment by sonyafenge** [comment]
close this one and replaced by PR #187

**Pull request closed without merging**